### DISCUSSION TITLE
请教下关于 onFormValuesChange 方法调用时机及频繁调用问题

### DISCUSSION THREAD

**Discussion by userhjp** [alibaba/formily#1847] [Q&A 2.x]
1、schemaJSON初始化时候会多次触发，当 schemaJSON 的 default有默认值时会导致 onFormValuesChange 触发，并且多次调用,例如以下schema在初始化会导致onFormValuesChange触发两次，我尝试删除其中一个字段的`default`则只触发了一次。
2、切换 display 会导致 onFormValuesChange 触发多次，如下代码，display 每次由 hidden 切换 visible 也会导致 onFormValuesChange 触发2次，总而言之，就是对象里面有几个属性则会触发几次。
请教下，以上问题是否正常现象？还是我代码写得有问题影响到了。
```
{
  type: object,
  'x-reactions': {
    dependencies: ['activeUUID'],
    fulfill: {
      state: {
        display: '{{$self.path.entire.endsWith($deps[0]) ? "visible" : "hidden"}}',
      },
    },
  },
  properties: {
    a: {
      type: 'number',
      default: 1,
      'x-decorator': 'FormItem',
      'x-component': 'NumberPicker',
    },
    b: {
      type: 'number',
      default: 2,
      'x-decorator': 'FormItem',
      'x-component': 'NumberPicker',
    },
  },
}
```

**Comment by janryWang** [comment]
display为none会触发删值，所以会触发valueChange事件

**Reply by userhjp** [reply]
那初始化时候呢？是正常的吗？我使用了FormTab 不知道是不是这个组件影响到了，不过问题不大，onFormValuesChange 没实际使用，目前只是开发时方便监听值，未其他用途，只是触发频率太高了没找到原因有点奇怪

**Reply by janryWang** [reply]
初始化也会触发删值的

**Reply by userhjp** [reply]
这样啊，那不管了，谢了！
